src/cbc_sdk/base.py

```diff
@@ -117,51 +145,129 @@ def __get__(self, instance, instance_type=None):
+
+        Args:
+            instance (NewBaseModel): Instance of the object to retrieve the field value from.
+            instance_type: TBD
+
+        Returns:
+            Any: Value of the field.
+        """
         ret = super(ArrayFieldDescriptor, self).__get__(instance, instance_type)
         return ret or []
 
 
 # TODO: this is a kludge to avoid writing "small" models?
 class ObjectFieldDescriptor(FieldDescriptor):
+    """Field descriptor for fields of 'object' type."""
     def __get__(self, instance, instance_type=None):
+        """
+        Retrieve the value of this field from an object.
+
+        Args:
+            instance (NewBaseModel): Instance of the object to retrieve the field value from.
+            instance_type: TBD
+
+        Returns:
+            Any: Value of the field.
+        """
         ret = super(ObjectFieldDescriptor, self).__get__(instance, instance_type)
         return ret or {}
 
 
 class IsoDateTimeFieldDescriptor(FieldDescriptor):
+    """Field descriptor for fields of 'iso-date-time' type."""
     def __init__(self, field_name):
+        """
+        Initialize the IsoDateTimeFieldDescriptor object.
+
+        Args:
+            field_name (str): The name of the field.
+        """
         super(IsoDateTimeFieldDescriptor, self).__init__(field_name)
 
     def __get__(self, instance, instance_type=None):
+        """
+        Retrieve the value of this field from an object.
+
+        Args:
+            instance (NewBaseModel): Instance of the object to retrieve the field value from.
+            instance_type: TBD
+
+        Returns:
+            Any: Value of the field.
+        """
         d = super(IsoDateTimeFieldDescriptor, self).__get__(instance, instance_type)
         return convert_from_cb(d)
 
     def __set__(self, instance, value):
+        """
+        Sets the value of this field on an object.
+
+        Args:
+            instance (NewBaseModel): Instance of the object to retrieve the field value from.
+            value (Any): New value for the field.
+        """
         parsed_date = convert_to_cb(value)
         super(IsoDateTimeFieldDescriptor, self).__set__(instance, parsed_date)
 
 
 class EpochDateTimeFieldDescriptor(FieldDescriptor):
+    """Field descriptor for fields of 'epoch-ms-date-time' type."""
     def __init__(self, field_name, multiplier=1.0):
+        """
+        Initialize the EpochDateTimeFieldDescriptor object.
+
+        Args:
+            field_name (str): The name of the field.
+            multiplier(float): Unused.
+        """
         super(EpochDateTimeFieldDescriptor, self).__init__(field_name)
         self.multiplier = float(multiplier)
 
     def __get__(self, instance, instance_type=None):
+        """
+        Retrieve the value of this field from an object.
+
+        Args:
+            instance (NewBaseModel): Instance of the object to retrieve the field value from.
+            instance_type: TBD
```

Comment:
Is this not used or do we need to figure out what it does? Same for all other instances of TBD

Reply:
This one I really don't know what it is.

Reply:
@avanbrunt-cb  helped resolve all TBDs.